client/src/main/kotlin/no/nav/su/se/bakover/client/person/AdGraphApiClient.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.su.se.bakover.client.person
+
+import arrow.core.Either
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.github.kittinunf.fuel.core.extensions.authentication
+import com.github.kittinunf.fuel.httpGet
+import no.nav.su.se.bakover.client.azure.OAuth
+import no.nav.su.se.bakover.common.objectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+data class AdGraphResponse(
+    val onPremisesSamAccountName: String,
+    val displayName: String,
+    val givenName: String,
+    val mail: String,
+    val officeLocation: String,
+    val surname: String,
+    val userPrincipalName: String,
+    val id: String,
+    val jobTitle: String
+)
+
+class AdGraphApiClient(
+    private val exchange: OAuth
+) {
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+
+    fun hent(userToken: String): Either<Pair<Int, String>, AdGraphResponse> {
+        val onBehalfOfToken = exchange.onBehalfOFToken(userToken, "https://graph.microsoft.com")
+        val query =
+            "onPremisesSamAccountName,displayName,givenName,mail,officeLocation,surname,userPrincipalName,id,jobTitle"
+        val graphUrl = "https://graph.microsoft.com/v1.0/me"
+
+        val (_, _, result) = graphUrl.httpGet(
+            listOf(
+                "\$select" to query
+            )
+        )
+            .authentication()
+            .bearer(onBehalfOfToken)
+            .header("Accept", "application/json")
+            .responseString()
+
+        return result.fold(
+            { json -> Either.right(objectMapper.readValue(json)) },
+            { error ->
+                log.info("Kall til Microsoft Graph API feilet", error)
+                val errorMessage = error.response.body().asString("application/json")
+                val statusCode = error.response.statusCode
+                Either.left(Pair(statusCode, errorMessage))
```

Comment:
Vet ikke om vi har noen "standard" for hvordan respons fra denne typen klienter ser ut?

Reply:
Vi har jo feilen i loggen. Så jeg bruker starte med å bare returnere et object som jeg definerer i interfacet. Vanligvis bør vi ikke returnere eksterne tjenesters feilkoder/meldinger til vår frontend.

Reply:
Det er som oftest `ClientResponse` som returneres tror jeg. Jeg syns det er litt synd, for da mister vi veldig mye informasjon; jeg syns det å transformere til `ClientResponse` heller hører hjemme i web-laget.

Reply:
Jeg har en langsiktig plan om å fjerne bruken av ClientResponse for alle klientene. Heller logge og lage object/data class, evt sealed class + objects/data classes. Har og en plan om å flytte alle client-interfacene inn i i domain. og legge disse feil-typene der. Enig med at web-laget bør selv velge status-koder og evt meldinger basert på disse feil-typene.